codegen/templates/endpoint.tmpl

```diff
@@ -126,6 +126,26 @@ func (h *{{$handlerName}}) HandleRequest(
 	{{$line}}
 	{{end}}
 
+	// log endpoint request to downstream services
+	zfields := []zapcore.Field{
+		zap.String("endpoint", h.endpoint.EndpointName),
+	}
+	{{if ne .RequestType ""}}
+	zfields = append(zfields, zap.String("body", fmt.Sprintf("%#v", requestBody)))
```

Comment:
`fmt.Sprintf("%#v", requestBody)` is reflection based and therefore could be expensive depending on the `requestBody`, ideally serialization should not happen until `Debug` is called, otherwise it's work done that is of no use. [zap.Object](https://godoc.org/go.uber.org/zap#Object) is better as marshaling is done lazily.

Reply:
The benchmark looks okay, we can think about `zap.Object` in a later time. Could you put a comment line like `// TODO: potential perf, use zap.Object lazy serialization` in here?

Also instead of  `fmt.Sprintf("%#v", requestBody))` we should just do `requestBody.String()`, the struct type generated by thrfitrw already implements `Stringer` interface.